src/main/kotlin/app/revanced/patches/tumblr/timelinefilter/TimelineFilterPatch.kt

```diff
@@ -0,0 +1,68 @@
+package app.revanced.patches.tumblr.timelinefilter
+
+import app.revanced.extensions.exception
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructionsWithLabels
+import app.revanced.patcher.extensions.InstructionExtensions.removeInstructions
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patches.tumblr.timelinefilter.fingerprints.PostsResponseConstructorFingerprint
+import app.revanced.patches.tumblr.timelinefilter.fingerprints.TimelineConstructorFingerprint
+import app.revanced.patches.tumblr.timelinefilter.fingerprints.TimelineFilterIntegrationFingerprint
+
+@Patch(description = "Filter what will be shown in the timeline.", requiresIntegrations = true)
+object TimelineFilterPatch : BytecodePatch(
+    setOf(TimelineConstructorFingerprint, TimelineFilterIntegrationFingerprint, PostsResponseConstructorFingerprint)
+) {
+    /**
+     * Add a filter to hide the given timeline object type.
+     * The list of all Timeline object types is found in the TimelineObjectType class,
+     * where they are mapped from their api name (returned by tumblr via the HTTP API) to the enum value name.
+     *
+     * @param typename The enum name of the timeline object type to hide.
+     */
+    @Suppress("KDocUnresolvedReference")
+    internal lateinit var addObjectTypeFilter: (typename: String) -> Unit private set
+
+    override fun execute(context: BytecodeContext) {
+
+        TimelineFilterIntegrationFingerprint.result?.let { integration ->
+            val startIndex = integration.scanResult.patternScanResult!!.startIndex
+
+            integration.mutableMethod.apply {
+                // Remove "BLOCKED_OBJECT_DUMMY" object type filter
+                removeInstructions(startIndex, 5)
```

Comment:
I have tried the approach of modifying (class member) static array instead, but could not get the compiler to output the array instantiation in a way that can be easily modified. Maybe because it is created via constant string primitives, but the static array is simply being generated as a bunch of `const-string` and then a single `filled-new-array`. This is of course not extendable since it needs more registers the more elements are added.

If you can find a way to actually implement this, that would be great - otherwise I don't know how this could be done without wrapping the filters into classes or fully re-implemeting the array instantiation in smali.

Reply:
I'm not sure how that would help. If you would like to try to get it working in a more elegant way than currently, feel free to - I've already tested multiple ways for multiple hours and have enough of fighting the java compiler simply because the patch-integration separation makes this so painfully difficult.

Reply:
I've added a warning comment. In the future if/when there are multiple types of filters, it could be refactored into wrapper classes for the filters since they would then have a proper purpose.

Reply:
> If you can find a way to actually implement this, that would be great.

If you can not abuse the compiler to generate what you want, you can do it manually via a patch.

1. Add a field to your mutable class:
2. Inject these instructions in your class initializer:

   ![image](https://github.com/ReVanced/revanced-patches/assets/13122796/31e638ce-95e7-4dce-a5dd-4d651b687eda)
3. Modify your `addFilter` delegate, so that it appends "array add item" instructions

Reply:
Then I will either have to inject a method (initializer) or fight with increasing its register count again if it doesn't have enough.

Is this required to get this merged?

Reply:
The class initializer is present in every compiled class, you just need to inject the bytecode to fill the array. It would indeed be better to do this as it would be equal to the other filter patches patch wise.

Reply:
That's how the litho filters are also initialized.

Reply:
Updated to use a static class member list and inject List.add("{elem}") calls instead